src/Illuminate/Support/Collection.php

```diff
@@ -201,9 +201,19 @@ public function first(callable $callback = null, $default = null)
      *
      * @return static
      */
-    public function flatten()
+    public function flatten($depth = INF)
```

Comment:
The problem with that is it messes up the recursion. The nice thing about `INF` is `INF - 1 === INF`, so you can unconditionally subtract 1 from the depth and have it all just work. Otherwise you need to special case check for `-1` or `null`, or whatever.